src/appMain/low_voltage_signals_handler.cc

```diff
@@ -0,0 +1,164 @@
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "low_voltage_signals_handler.h"
+#include <signal.h>
+#include "life_cycle.h"
+#include "utils/logger.h"
+#include "config_profile/profile.h"
+
+namespace main_namespace {
+
+namespace {
+// Generic function to be able to log enum class values
+template <typename T>
+std::ostream& operator<<(
+    typename std::enable_if<std::is_enum<T>::value, std::ostream>::type& stream,
+    const T& e) {
+  return stream << static_cast<typename std::underlying_type<T>::type>(e);
+}
+}  // namespace
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "LowVoltageSignalsHandler")
+
+LowVoltageSignalsHandler::LowVoltageSignalsHandler(
+    LifeCycle& life_cycle, const LowVoltageSignalsOffset& offset_data)
+    : state_(SDLState::kRun)
+    , notifications_delegate_(nullptr)
+    , life_cycle_(life_cycle)
+    , signals_handler_thread_(nullptr)
+    , SIGLOWVOLTAGE_(offset_data.low_voltage_signal_offset + SIGRTMIN)
+    , SIGWAKEUP_(offset_data.wake_up_signal_offset + SIGRTMIN)
+    , SIGIGNOFF_(offset_data.ignition_off_signal_offset + SIGRTMIN) {}
+
+bool LowVoltageSignalsHandler::Init() {
+  notifications_delegate_ = new NotificationThreadDelegate(*this);
+  signals_handler_thread_ = threads::CreateThread("LV_SIGNALS_HANDLER_THREAD",
+                                                  notifications_delegate_);
+  signals_handler_thread_->start();
+  return true;
+}
+
+SDLState LowVoltageSignalsHandler::get_current_sdl_state() const {
+  return state_;
+}
+
+int LowVoltageSignalsHandler::low_voltage_signo() const {
+  return SIGLOWVOLTAGE_;
+}
+
+int LowVoltageSignalsHandler::wake_up_signo() const {
+  return SIGWAKEUP_;
+}
+
+int LowVoltageSignalsHandler::ignition_off_signo() const {
+  return SIGIGNOFF_;
+}
+
+void LowVoltageSignalsHandler::Destroy() {
+  state_ = SDLState::kStop;
+  if (signals_handler_thread_) {
+    signals_handler_thread_->join();
+  }
+  delete notifications_delegate_;
```

Comment:
@AByzhynar Please check the null pointer before deleting.

Reply:
@BSolonenko What for?